silver/utils/decorators.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2016 Presslabs SRL
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from django.core.cache import cache
+
+
+def cached_method(unique_attribute=None):
+    """
+    A decorator for caching method results in the Django context
+    :param unique_attribute: Name of attribute used in the cache key
+    """
+    def decorator(func):
+        def func_wrapper(self, *args, **kwargs):
+            """
+            :param invalidate_cache: If True the cache is being invalidated
+            :param set_cache: If True the cache will be set if there is no cache
+                              If invalidate is True, the cache will be reset
+            :param bypass_cache: If True the original method will be called
+                                 It can interact with the other parameters
+            :param return: Value of method (either cached or not)
+
+            !!! Warning: the *args and **kwargs parameters are considered when
+            creating the cache key
+            """
+
+            invalidate = kwargs.pop('invalidate_cache', False)
+            set_cache = kwargs.pop('set_cache', True)
+            bypass = kwargs.pop('bypass_cache', False)
+
+            if not unique_attribute:
+                unique_key = id(self)
```

Comment:
Usually a hash function is called. The id of the object is it's identifier, during the lifetime of the object. Each class that inherit from `object` should have an `__hash__` method. I suggest to use that instead of `id`